test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala

```diff
@@ -0,0 +1,248 @@
+        Json.parse(
+          """
+            |{
+            | "blah" : "blah"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.addFormIdToWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
+        Future.successful(
+          HttpResponse(Http.Status.NOT_FOUND)
+        )
+      )
+      private val result = whitelistController.confirmAdd()(addToken(fakeRequestWithBody))
+      status(result) shouldBe Status.BAD_REQUEST
+    }
+  }
+
+  "confirmDelete" should {
+
+    "return 303 (Redirect) when a Form ID is successfully deleted via the message service" in new WhitelistControllerTestCase {
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      private val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "formId" : "SA316 2015",
+            | "reasonText" : "some reason text"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.deleteFormIdFromWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
+        Future.successful(
+          HttpResponse(Http.Status.OK)
+        )
+      )
+      private val result = whitelistController.confirmDelete()(addToken(fakeRequestWithBody))
+      status(result) shouldBe Status.SEE_OTHER
+    }
+
+    "return 502 (Bad Gateway) when the message service returns any other status" in new WhitelistControllerTestCase {
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      private val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "formId" : "SA316 2015",
+            | "reasonText" : "some reason text"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.deleteFormIdFromWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
+        Future.successful(
+          HttpResponse(Http.Status.NOT_FOUND)
+        )
+      )
+      private val result = whitelistController.confirmDelete()(addToken(fakeRequestWithBody))
+      status(result) shouldBe Status.BAD_GATEWAY
+    }
+
+    "return 400 (Bad Request) when the form ID JSON is not valid" in new WhitelistControllerTestCase {
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "blah" : "blah"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.deleteFormIdFromWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
```

Comment:
Same here, think maybe the controller would throw the error before it reaches the connector. Maybe not needed?

Reply:
Yep, you're right @elabeca. Now removed.